feg/gateway/services/n7_n40_proxy/n7/model_conversion.go

```diff
@@ -0,0 +1,567 @@
+	pccRule *n7_sbi.PccRule,
+) *protos.PolicyRule {
+	// For all Ref*Data Maximum one entry is expected.
+	// These are declared as arrays for future compatibility
+	policyRule := &protos.PolicyRule{
+		Id:            pccRule.PccRuleId,
+		Priority:      getSbiUinteger(pccRule.Precedence),
+		MonitoringKey: getMonKeyFromPccRule(policyDecision, pccRule),
+		Redirect:      getTcDataFromPccRule(policyDecision, pccRule),
+		Qos:           getQosDataFromPccRule(policyDecision, pccRule),
+		FlowList:      getFlowListFromPccRule(pccRule),
+	}
+	if pccRule.RefChgData != nil && len(*pccRule.RefChgData) > 0 && policyDecision.ChgDecs != nil {
+		chgData, found := policyDecision.ChgDecs.Get((*pccRule.RefChgData)[0])
+		if found {
+			policyRule.RatingGroup = uint32(*chgData.RatingGroup)
+			policyRule.ServiceIdentifier = &protos.ServiceIdentifier{
+				Value: uint32(*chgData.ServiceId),
+			}
+			policyRule.Offline = getSbiBool(chgData.Offline)
+			policyRule.Online = getSbiBool(chgData.Online)
+		}
+	}
+	policyRule.TrackingType = getTrackingTypeProto(len(policyRule.MonitoringKey) != 0, policyRule.RatingGroup != 0)
+
+	if len(policyRule.MonitoringKey) == 0 && policyRule.Redirect == nil &&
+		policyRule.Qos == nil && len(policyRule.FlowList) == 0 && policyRule.RatingGroup == 0 {
+		// must be a static rule if these are not present
+		return nil
+	}
+	return policyRule
+}
+
+func getMonKeyFromPccRule(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) []byte {
+	if pccRule.RefUmData != nil && len(*pccRule.RefUmData) > 0 && policyDecision.UmDecs != nil {
+		umData, found := policyDecision.UmDecs.Get((*pccRule.RefUmData)[0])
+		if found {
+			return []byte(umData.UmId)
+		}
+	}
+	return []byte{}
+}
+
+func getTcDataFromPccRule(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) *protos.RedirectInformation {
+	if pccRule.RefTcData != nil && len(*pccRule.RefTcData) > 0 && policyDecision.TraffContDecs != nil {
+		tcData, found := policyDecision.TraffContDecs.Get((*pccRule.RefTcData)[0])
+		if found {
+			return GetRedirectInfoProto(tcData.RedirectInfo)
+		}
+	}
+	return nil
+}
+
+func getQosDataFromPccRule(
+	policyDecision *n7_sbi.SmPolicyDecision,
+	pccRule *n7_sbi.PccRule,
+) *protos.FlowQos {
+	if pccRule.RefQosData != nil && len(*pccRule.RefQosData) > 0 && policyDecision.QosDecs != nil {
+		qosData, found := policyDecision.QosDecs.Get((*pccRule.RefQosData)[0])
+		if found {
+			return GetFlowQosProto(&qosData)
+		}
+	}
+	return nil
+}
+
+func getFlowListFromPccRule(pccRule *n7_sbi.PccRule) []*protos.FlowDescription {
+	var flowList []*protos.FlowDescription
+	if pccRule.FlowInfos != nil {
+		for _, flowInfo := range *pccRule.FlowInfos {
+			if flowInfo.FlowDescription != nil {
+				flow, err := policydb.GetFlowDescriptionFromFlowString(string(*flowInfo.FlowDescription))
+				if err != nil {
+					glog.Errorf("Could not get flow for description %s : %s", *flowInfo.FlowDescription, err)
```

Comment:
this error is still not handled. 
What will happen to the rule in sessiond if it is sent without any flow description? Will this rule be valid?

Reply:
Since this is received in the answer response, will not be able to inform PCF about the error. This behavior was taken from [here](https://github.com/magma/magma/blob/f9840abd66b4616695b7ec6405ae87186c65c516/feg/gateway/services/session_proxy/credit_control/gx/model_conversion.go#L155).

Reply:
I see... not good at all. But lets leave it as it is then

In my opinion we should error this one and not process it